lib/diagnostic.js

```diff
@@ -24,7 +34,13 @@ const sum = function sum() {
 // if invoking the callback for any element
 // in the array returns a truthy value
 const any = function any(array, callback) {
-
+  for (let i = 0; i < array.length; i++) {
+    if(callback(array[i])){
+      return true;
+    } else {
+      return false;
+    }
+  }
```

Comment:
coz it will exit the loop the minute it hit false.  so it will not be able to go thru the whole array.

> On Jan 27, 2016, at 12:04 PM, Matt Brendzel <notifications@github.com> wrote:
> 
> In lib/diagnostic.js <https://github.com/ga-wdi-boston/js-array-methods-diagnostic/pull/49#discussion_r51014638>:
> 
> > @@ -24,7 +34,13 @@ const sum = function sum() {
> >  // if invoking the callback for any element
> >  // in the array returns a truthy value
> >  const any = function any(array, callback) {
> > -
> > +  for (let i = 0; i < array.length; i++) {
> > +    if(callback(array[i])){
> > +      return true;
> > +    } else {
> > +      return false;
> > +    }
> > +  }
> Almost. This part is correct:
> 
> if(callback(array[i])){
>      return true;
> }
> But the else breaks it. Can you see why?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ga-wdi-boston/js-array-methods-diagnostic/pull/49/files#r51014638>.
>